crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -76,19 +86,29 @@ def adminrepocheck
 
   def adminbackup
     upgrade_status = ::Crowbar::UpgradeStatus.new
-    upgrade_status.start_step if upgrade_status.current_step == :admin_backup
+    upgrade_status.start_step(:admin_backup)
     @backup = Backup.new(backup_params)
 
     if @backup.save
-      upgrade_status.end_step if upgrade_status.current_step == :admin_backup
+      upgrade_status.end_step
       render json: @backup, status: :ok
     else
       upgrade_status.end_step(
         false,
         admin_backup: @backup.errors.full_messages.first
-      ) if upgrade_status.current_step == :admin_backup
+      )
       render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
     end
+  rescue Crowbar::UpgradeStatus::StartStepRunningError,
+         Crowbar::UpgradeStatus::StartStepOrderError => e
```

Comment:
i don't think so, or will it ever happen at that stage that the step is not running?